### PR TITLE
create all dto and mapper

### PR DESCRIPTION
**Pull request by smazurok147**

### CODE REVIEW

File: BACKEND/src/main/java/com/example/dto/product/ProductRequest.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.dto.product;
+
+import lombok.Data;
+
+@Data
+public class ProductRequest {
+    private String name;
+    private Integer price;
+}
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/product/ProductRequest.java]: add field description

File: BACKEND/src/main/java/com/example/dto/orderitem/OrderItemRequest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@Data
+
+public class OrderItemRequest {
+
+    private Products product;
+    private Short quantity;
+
+
+}
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/orderitem/OrderItemRequest.java]:
```
    private Integer productId;
    private Short quantity;
```

File: BACKEND/src/main/java/com/example/dto/orderitem/OrderItemResponse.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.example.dto.orderitem;
+
+import com.example.entity.Products;
+import lombok.Data;
+
+@Data
+public class OrderItemResponse {
+    private Integer id;
+    private Products product;
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/orderitem/OrderItemResponse.java]: `private Products product;` -> `private ProductResponse product`

File: BACKEND/src/main/java/com/example/dto/order/OrderRequest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Data;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Data
+public class OrderRequest {
+    private User user;
+    private List<OrderItems> orderItems;
+    private LocalDateTime createdAt;
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/order/OrderRequest.java]:
```
private Integer userId;
private List<OrderItemsRequest> orderItems;
```

File: BACKEND/src/main/java/com/example/dto/order/OrderResponse.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+@Data
+public class OrderResponse {
+    private Integer id;
+    private User user;
+    private List<OrderItems> orderItems;
+    private Float total_amount;
+    private String status;
+    private LocalDateTime createdAt;
+}
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/order/OrderResponse.java]:
```
    private Integer id;
    private UserResponse user;
    private List<OrderItemsResponse> orderItems;
    private Float total_amount;
    private String status;
    private LocalDateTime createdAt;
```

File: BACKEND/src/main/java/com/example/dto/user/UserRequest.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.example.dto.user;
+
+import com.example.entity.Role;
+import lombok.Data;
+
+@Data
+public class UserRequest {
+    private String username;
+    private String password;
+    private Role role;
```
Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/user/UserRequest.java]: delete role from request

Comment (smazurok147) [inline BACKEND/src/main/java/com/example/dto/user/UserRequest.java]: add email

File: BACKEND/src/main/java/com/example/mapper/BaseConverter.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+public abstract class BaseConverter<Entity,Request,Response> {
+    public abstract Entity requestToEntity(Request request);
+    public abstract Response entityToResponse(Entity entity);
+    public abstract List<Response> entityToResponse(List<Entity> entity);
+
+    public abstract Response entityToResponse(Optional<Entity> byId);
```
Comment (Pixel147) [inline BACKEND/src/main/java/com/example/mapper/BaseConverter.java:11]: public abstract Response entityToResponse(Entity entity);

Comment (smazurok147) [inline BACKEND/src/main/java/com/example/mapper/BaseConverter.java]: delete this

**Pull request merged**